src/main/java/net/fabricmc/mappingio/MappingReader.java

```diff
@@ -208,10 +211,23 @@ public static List<String> getNamespaces(Reader reader, MappingFormat format) th
 	 * @param visitor The receiving visitor.
 	 * @throws IOException If the format can't be detected or reading fails.
 	 */
+	@Deprecated
```

Comment:
Why deprecate this, I think by default it should fail when it reads a malformed mapping file?

Reply:
Is ther why `new ThrowingErrorSink(Severity.WARNING)` is a bad default severity? Overloads is like a way to imply default arguments.

Reply:
It hinders readability and is a pain to maintain. We already have up to five overloads per mapping reader, imagine we were to add an additional argument in the future, it would double the count ten method definitions. `MappingReader#read` already has *eight*, which I'm pretty sure starts getting confusing for consumers

Reply:
If we have too many optional arguments, we might just use builders.